keras/backend/tensorflow_backend.py

```diff
@@ -2623,6 +2623,13 @@ def _step(time, output_ta_t, *states):
         new_states = final_outputs[2:]
 
         outputs = output_ta.stack()
+
+        # TF while_loop removed 0-th dimension from the outputs
+        # but we actually know the expected shape of the output
+        output_shape = list(outputs.get_shape())
+        output_shape[0] = inputs.get_shape()[0]
+        outputs.set_shape(output_shape)
```

Comment:
You can replace these lines with `outputs.set_shape((inputs.get_shape()[0], None, None))`